src/backend/routers/conversation.py

```diff
@@ -363,32 +433,19 @@ async def batch_upload_file(
             )
 
     # TODO: check if file already exists in DB once we have files per agents
-
-    # Handle uploading File
-    files_to_upload = []
-    for file in files:
-        content = await get_file_content(file)
-        cleaned_content = content.replace("\x00", "")
-        filename = file.filename.encode("ascii", "ignore").decode("utf-8")
-
-        # Create File
-        upload_file = FileModel(
-            user_id=conversation.user_id,
-            conversation_id=conversation.id,
-            file_name=filename,
-            file_path=filename,
-            file_size=file.size,
-            file_content=cleaned_content,
-        )
-        files_to_upload.append(upload_file)
     try:
-        uploaded_files = file_crud.batch_create_files(session, files_to_upload)
+        uploaded_files = await file_service.create_conversation_files(
+            session, files, user_id, conversation.id
+        )
     except Exception as e:
         raise HTTPException(
             status_code=500, detail=f"Error while uploading file(s): {e}."
         )
 
-    return uploaded_files
+    files_with_conversation_id = attach_conversation_id_to_file(
```

Comment:
we do this because files dont have a convo id where its stored now (goal to be agnostic to BE code). however, the endpoints and FE still needs convo id with the file. can potentially always return with file ID in the file service methods, but those methods are not always used for router logic.